git_creation_and_tracking/git_creation_and_tracking.md

```diff
@@ -0,0 +1,829 @@
+the commit's full identifier
+(which starts with the same characters as
+the short identifier printed by the `git commit` command earlier),
+the commit's author,
+when it was created,
+and the log message Git was given when the commit was created.
+
+<div class = 'help'>
+**Where Are My Changes?** If we run `ls` at this point, we will still see just one file called `mars.txt`. That's because Git saves information about files' history in the special `.git` directory mentioned earlier so that our filesystem doesn't become cluttered (and so that we can't accidentally edit or delete an old version).
+</div>
+
+Now suppose Dracula adds more information to the file.
+(Again, we'll edit with `nano` and then `cat` the file to show its contents;
+you may use a different editor, and don't need to `cat`.)
+
+```
+$ nano mars.txt
+$ cat mars.txt
+
+Cold and dry, but everything is my favorite color
+The two moons may be a problem for Wolfman
+```
+
+When we run `git status` now,
+it tells us that a file it already knows about has been modified:
+
+```
+$ git status
+
+On branch main
+Changes not staged for commit:
+  (use "git add <file>..." to update what will be committed)
+  (use "git checkout -- <file>..." to discard changes in working directory)
+
+	modified:   mars.txt
+
+no changes added to commit (use "git add" and/or "git commit -a")
+```
+
+The last line is the key phrase:
+"no changes added to commit".
+We have changed this file,
+but we haven't told Git we will want to track those changes
+(which we do with `git add`)
+nor have we recorded them (which we do with `git commit`).
+So let's do that now. It is good practice to always review
+our changes before recording them. We do this using `git diff`.
+This shows us the differences between the current state
+of the file and the most recently saved version:
+
+```
+$ git diff
+
+diff --git a/mars.txt b/mars.txt
+index df0654a..315bf3a 100644
+--- a/mars.txt
++++ b/mars.txt
+@@ -1 +1,2 @@
+ Cold and dry, but everything is my favorite color
++The two moons may be a problem for Wolfman
+```
+
+<div class = 'care'>
+The output is cryptic because it is meant for a computer, not a human.
+It is actually a series of commands for tools like editors and `patch`
+telling them how to reconstruct one file given the other. Nevertheless we can gain some useful information
+if we break it down into pieces:
+
+  1. The first line tells us that Git is comparing the old and new versions of the file `mars.txt`.
+  2. The second line tells exactly which versions of the file Git is comparing; `df0654a` and `315bf3a` are unique computer-generated labels for those versions.
+  3. The third and fourth lines once again show the name of the file being changed.
+  4. The remaining lines are the most interesting, they show us the actual differences and the lines on which they occur. In particular, the `+` marker in the first column shows where we added a line.
+</div>
+
+<div class = 'warning'>
```

Comment:
Do you think it would be worth turning this into a quick quiz question? It already fits that format perfectly!